pkg/bmc/access.go

```diff
@@ -128,3 +136,75 @@ func NewAccessDetails(address string, disableCertificateVerification bool) (Acce
 
 	return factory(parsedURL, disableCertificateVerification)
 }
+
+// A private method for building firmware config to BIOS settings for different driver.
+// NOTE： firmwareConfig can't be a pointer
+// NOTE: This function doesn't handle nested structure
+func buildBIOSSettings(firmwareConfig interface{}, exclude []string, nameMap map[string]string, valueMap map[string]string) (settings []map[string]string, err error) {
```

Comment:
This function seems worse than just hard-coding the implementation for each driver type. What is the problem we're trying to solve here?

Reply:
Since there are many BIOS settings provided in the previous code, this function was created in order to avoid using too much `if`. Because the settings have been reduced now, there is no value anymore.